### ISSUE TITLE
Simplify Imitation Learning Pipeline with Offline Demonstrations

### ISSUE THREAD

**Issue by Jybbs** [Jybbs/Thermur#46]
The current training infrastructure uses PyTorch Lightning with TorchRL environments, implementing online demonstration collection for an expert controller that never changes. Since the MurmurationController is a stable, fixed component, we're unnecessarily regenerating identical demonstrations every training run, burning compute and maintaining complex infrastructure for no benefit.

By recognizing that our expert is static and moving to offline demonstration generation with PyG's native Lightning integration, we can eliminate ~70% of the training boilerplate, train 10x faster, and achieve better reproducibility.

This migration warrants bumping to v0.2.0 as it fundamentally changes the data pipeline architecture while preserving all scaling capabilities.

### Problem Analysis

The current architecture regenerates identical expert demonstrations every training run because we treat the MurmurationController as if it needs online interaction. This creates several inefficiencies:

**Compute Waste**: Each training run spends significant time generating the same demonstrations from the same expert controller with the same WRF scenarios.

**Complex Infrastructure**: We maintain TorchRL environments, TensorDict replay buffers, and experience collectors for what is essentially static data generation.

**Slow Iteration**: Experimenting with different architectures or hyperparameters requires waiting for demonstration generation each time.

**Debugging Difficulty**: Can't inspect training data before training starts, making it harder to validate the expert's behavior.

### Proposed Solution

Implement a two-phase workflow that separates demonstration generation from training:

**Phase 1: Demonstration Generation** (run once)
```bash
thermur demonstrations generate
```
Generate expert demonstrations offline using the MurmurationController directly, saving them as PyG Data objects for reuse across multiple training runs.

**Phase 2: Training** (run many times)
```bash
thermur train
```
Train on pre-generated demonstrations with instant startup, enabling rapid experimentation with architectures and hyperparameters.

### Technical Design

The refactoring focuses on three core changes: making the controller output PyG data directly, creating offline demonstration storage, and simplifying the data pipeline.

**Expert Controller PyG Integration**

Extend `MurmurationController` with a method to generate trajectories as PyG Data objects. This method will initialize flock state, iterate through timesteps while querying WRF data, compute graph connectivity using existing `_compute_topological_neighbors()`, and create PyG Data objects with the same 13D feature vector the GNN expects. The implementation should reuse all existing physics and safety logic.

**Demonstrations Module**

Create a new module for managing offline demonstrations with Pydantic configuration for paths and parameters. The module will auto-discover WRF scenarios, generate demonstrations using the expert controller, split data into train/validation sets, and save as `.pt` files with metadata for validation. Rich progress tracking will provide feedback during generation.

**Simplified Data Pipeline**

Replace the complex TensorDict replay buffer with a minimal PyG Dataset that loads pre-generated graphs. The new DataModule will use PyG's `LightningDataset` for optimized batching, automatic multi-GPU support, and efficient memory management. This reduces the data pipeline from ~1000 lines to ~300 lines.

**CLI Integration**

Add a `demonstrations` command that generates expert data with Rich progress bars and fire gradient styling. The training command will auto-detect demonstrations or prompt to generate them if missing. Both commands will use the existing ThermurUI for consistent output formatting.

#### Phase 1: Controller Enhancement
- [x] Add `generate_trajectory_as_pyg()` method to `MurmurationController`
- [x] Add helper method for single timestep PyG creation
- [x] Test trajectory generation with existing WRF data
- [x] Verify feature dimensions match GNN expectations (13D)
- [x] Ensure edge computation uses existing methods

#### Phase 2: Data Infrastructure
- [x] Create `DemonstrationsModel` in `config/imitation/data/schemas.py`
- [x] Implement `Demonstrations` class with generation/storage logic
- [x] Add metadata tracking (timestamps, counts, checksums)
- [x] Create validation methods for data integrity
- [x] Test save/load cycle with PyG Data objects

#### Phase 3: Dataset and DataModule
- [x] Create minimal `ThermurDataset` extending PyG's `Dataset`
- [x] Replace `experience.py` with simplified `DataModule`
- [x] Integrate PyG's `LightningDataset` for batching
- [x] Remove TensorDict replay buffer code
- [x] Test dataloader performance and memory usage

#### Phase 4: CLI Integration
- [ ] Add `demonstrations` command to CLI app
- [x] Implement Rich progress tracking
- [x] Add fire gradient styling for output
- [ ] Create validation subcommand
- [x] Update `train` command for auto-detection
- [x] Add prompts for missing demonstrations

#### Phase 5: Training Pipeline
- [x] Update training initialization to use new DataModule
- [x] Remove environment and collector setup
- [x] Add optional torch.compile to GNNPolicy
- [x] Verify end-to-end training with new pipeline
- [x] Test multi-GPU training compatibility

#### Phase 6: Cleanup and Migration
- [x] Delete `environment.py`
- [x] Delete old `experience.py` and `collector.py`
- [x] Remove `torchrl` and `tensordict` from `pyproject.toml`
- [x] Update all import statements
- [x] Remove unused config sections
- [x] Update documentation

#### Phase 7: Testing and Validation
- [x] Benchmark training speed improvements
- [x] Verify reproducibility with fixed demonstrations
- [x] Test memory usage reduction
- [x] Validate multi-GPU scaling
- [x] Ensure all existing tests pass
- [x] Add new tests for demonstrations module

#### Phase 8: Upgrade Package Version
- [x] Update version in TOML
- [x] Ensure updated version is retrievable in `thermur info`
- [ ] Design a "Releases" pattern in Github

### Acceptance Criteria

- 70% reduction in data pipeline code (from ~1000 to ~300 lines)
- Demonstrations can be generated and saved offline
- Training works with pre-generated demonstrations
- Complete removal of TorchRL dependency
- Complete removal of TensorDict dependency
- Direct PyG data flow: controller → dataset → policy
- 10x faster training iteration (no online generation)
- Multi-GPU training remains functional
- All existing tests pass with new architecture
- Rich CLI output with progress tracking
- Metadata validation for demonstrations
